velox/exec/RowContainer.cpp

```diff
@@ -512,4 +504,244 @@ void RowContainer::setProbedFlag(char** rows, int32_t numRows) {
   }
 }
 
+RowTypePtr RowContainer::spillType() {
+  if (spillType_) {
+    return spillType_;
+  }
+  std::vector<std::string> names;
+  std::vector<TypePtr> types = keyTypes_;
+  for (auto& aggregate : aggregates_) {
+    types.push_back(aggregate->accumulatorType());
+  }
+  types.insert(types.end(), types_.begin() + keyTypes_.size(), types_.end());
+  for (auto i = 0; i < types.size(); ++i) {
+    names.push_back(fmt::format("c{}", i));
+  }
+  return spillType_ = ROW(std::move(names), std::move(types));
+}
+
+void RowContainer::extractSpill(folly::Range<char**> rows, RowVector& result) {
+  result.resize(rows.size());
+  for (auto i = 0; i < keyTypes_.size(); ++i) {
+    extractColumn(rows.data(), rows.size(), i, result.childAt(i));
+  }
+  for (auto i = 0; i < aggregates_.size(); ++i) {
```

Comment:
It appears as if many comments got lost. I'm copying one here and will pause reviewing until I hear back.

---
It bothers me that aggregation and join-specific logic is intertwined with otherwise generic RowContainer code. I wonder if the logic in the method would be better placed into aggregation and join operators. I think this would make more sense as there is nothing spill-specific in this method other than the fact that this logic is used in spilling workflow.

Perhaps, add some comments here to make it clear to the reader that some parts of this method are for join and some for group by, e.g.

```
if (aggregates_.empty()) {
   // Extract dependent columns for the join hash table.
   ...
} else {
  // Extract group-by accumulator state.
 ...
}
```